roles/iridl/templates/update_datalib

```diff
@@ -25,37 +37,49 @@ init_git() {
         fail $? "failed to checkout version $version"
     git submodule update --init --recursive || fail $? "failed to update submodules"
 
+    if [ ! -z "${clean}" ]; then
+        git clean -fdx || fail $? "git clean failed"
+        git submodule foreach --recursive git clean -fdx || fail $? "submodule clean failed"
+    fi
 }
 
 if [[ $(id -u) != 0 ]]; then
     fail 1 "This script should be run with sudo."
 fi
 
+# properly set interactive flags if in an interactive for any docker commands we run
+if [[ -t 1 ]]; then
+    interactive_args=-it
+fi
 
 ### dlentries
 
 if [[ ! -z "{{dlentries_repo}}" ]]; then
-    init_git {{dlentries_repo}} {{dlentries_dir}} {{dlentries_version}}
+    init_git {{dlentries_repo}} {{dlentries_dir}} {{dlentries_version}} /usr/local/bin/ssh-git
 fi
 
-### maproom
+### python maproom
 
-init_git {{maproom_repo}} {{maproom_src_dir}} {{maproom_version}}
+# only if we actually have a python maproom for this deployment
+{% if python_maproom_repo is defined %}
+init_git {{python_maproom_repo}} {python_maproom_src_dir}} \
+         {{python_maproom_version}} {{python_maproom_git_ssh}} clean
 
-# cleanup beforehand
-git clean -fdx || fail $? "git clean failed"
-git submodule foreach --recursive git clean -fdx || fail $? "submodule clean failed"
+docker build -t python_maproom:{{python_maproom_version}} .
```

Comment:
Docker actually has the ability to build directly from a git repository. While we could use that, I feel a little funny about not maintaining more fine grained control over the process. Also, there may be circumstances where images may need resources that aren't in the repository (like custom assets)